addons/controls/README.md

```diff
@@ -0,0 +1,230 @@
+- [Writing stories](#writing-stories)
+  - [Auto-generated args](#auto-generated-args)
+  - [Manually-configured args](#manually-configured-args)
+- [Migrating from knobs](#migrating-from-knobs)
+  - [Knobs to auto-generated args](#knobs-to-auto-generated-args)
+  - [Knobs to manually-configured args](#knobs-to-manually-configured-args)
+- [Available controls](#available-controls)
+- [Parameters](#parameters)
+    - [expanded](#expanded)
+- [FAQs](#faqs)
+  - [How will this replace addon-knobs?](#how-will-this-replace-addon-knobs)
+
+## Get started
+
+To get started with `addon-controls`:
+
+1. [install the addon](#installation)
+2. [write an args story](#writing-stories)
+3. [configure the addon (optional)](#configuration)
+
+If you're installing it into an existing project that uses `addon-knobs`, you might also [migrate your existing stories](#migrating-from-knobs).
+
+## Installation
+
+In the first version, Controls requires that you're using [Storybook Docs](https://github.com/storybookjs/storybook/tree/master/addons/docs). If you're not using it already, please install that first.
+
+Next, install the package:
+
+```sh
+npm install @storybook/addon-controls -D # or yarn
+```
+
+And add it to your `.storybook/main.js` config:
+
+```js
+module.exports = {
+  addons: ['@storybook/addon-controls'],
+};
+```
+
+Once the addon is installed you should see a `Controls` tab in the addons panel:
+
+## Writing stories
+
+Controls are only available for stories that make use of [Storybook Args](https://github.com/storybookjs/storybook/blob/next/docs/src/pages/formats/component-story-format/index.md#args-story-inputs), so if your story functions don't accept an `Args` object as the first argument, you'll need to upgrade your stories.
+
+Here's a `react` story that doesn't use args:
+
+```js
+export const Basic = () => <Button label="hello" />;
+```
+
+And here's one that does:
+
+```js
+export const Basic = (args = { label: 'hello' }) => <Button {...args} />;
+```
+
+There are a few different ways to get from here to there:
+
+- The quickest way to do this is to use [auto-generated args](#auto-generated-args).
+- If you're implementing custom story logic based on control value, you should use [manually-configured args](#manually-configured-args)
+- And if you're adding this to an existing project that's already using knobs, you should [migrate it](#migrating-from-knobs).
+
+### Auto-generated args
+
+In the example above, [Storybook Docs](https://github.com/storybookjs/storybook/tree/next/addons/docs) automatically annotates the `Button` with `ArgTypes` metadata that allows `Controls` to automatically create rows for that component.
+
+So if the full story was:
+
+```js
+import { Button } from './Button';
+export default { title: 'Button', component: Button };
+
+export const Basic = () => <Button label="hello" />;
+```
+
+You could simply rewrite it as:
+
+```js
+export const Basic = (args = { label: 'hello' }) => <Button {...args} />;
```

Comment:
Don't forget to add `import { Button } from './Button';` to this example too

Reply:
It just makes the page much longer without providing informational value. WDYT?

Reply:
I'm not sure, the docs talk about automatic stuff and you're showing scenario A to B, where in scenario A you have an import and in scenario B you don't.. so it might be confusing for people without context. I'd say either:
1 - Remove the import from both, people can infer that the Button is coming from somewhere
2 - Make the text more specific like: `Rewrite the Basic story as:`